api/app/jobs/noaa.py

```diff
@@ -55,6 +67,66 @@ def get_year_mo_date_string_from_datetime(datetime: datetime.datetime) -> str:
     return year_mo_date
 
 
+def get_noaa_wx_variables_filter_str() -> str:
+    wx_vars_filter_str = ''
+    for var in WX_VARS:
+        wx_vars_filter_str += f'var_{var}=on&'
+
+    return wx_vars_filter_str
+
+
+def get_noaa_levels_filter_str() -> str:
+    levels_filter_str = ''
+    for level in LEVELS:
+        levels_filter_str += f'lev_{level}=on&'
+
+    return levels_filter_str
+
+
+def get_noaa_subregion_filter_str() -> str:
+    return f'subregion=&toplat={SUBREGION_TOP_LAT}&leftlon={SUBREGION_LEFT_LON}&rightlon={SUBREGION_RIGHT_LON}'\
+        f'&bottomlat={SUBREGION_BOTTOM_LAT}'
+
+
+def get_nam_model_run_download_urls(download_date: datetime.datetime, model_cycle: str) -> Generator[str, None, None]:
+    """ Yield URLs to download NAM North America model runs """
+    if model_cycle == '00':
+        noon = [20]
```

Comment:
I assume this is the 20th hour, but what are the other values, e.g. 45, 63?

Reply:
Added a comment to document why those hours were chosen for each model cycle

Reply:
Is an hour past 24 sometime in the next day(s)?

Reply:
@conbrad based on your question I'm thinking maybe GitHub has a stale cache again? The comment I added with documentation isn't showing up in the Review Changes view, but it's here - https://github.com/bcgov/wps/pull/2790/commits/822e50a374915479fbe0ac2e32e29b33c4752ce7

Reply:
I see this:

```
    # for model_cycle 00:
    # for first day, need hour 20:00 UTC (13:00 PST)
    # Day 2: need hours 42 and 45 to interpolate for hour 44 (13:00 PST)
    # Day 3: need hours 66 and 69 to interpolate for hour 68 (13:00 PST)
    #
    # for model_cycle 06:
    # for first day, need hour 14 (14+6 = 20:00 UTC)
    # Day 2: need hours 36 and 39 to interpolate for hour 38 (13:00 PST)
    # Day 3: need hours 60 and 63 to interpolate for hour 62 (13:00 PST)
    #
    # for model_cycle 12:
    # for first day, need hour 8 (12+8 = 20:00 UTC)
    # for second day, need hour 32 (12 + (32-24) = 20:00 UTC)
    # Day 3: need hours 54 and 57 to interpolate for hour 56 (13:00 PST)
    #
    # for model_cycle 18:
    # for first day, need hour 2 (18 + 2 = 20:00 UTC)
    # for second day, need hour 26 (18 + (26-24) = 20:00 UTC)
    # Day 3: need hours 48 and 51 to interpolate for hour 50 (13:00 PST)
   ```
   If that's not up to date then yeah it's a caching issue, but if it is I have no idea what hours beyond 24 mean, I'm assuming they're hours offset from some start date past a day.

Reply:
yep that's what I was referring to. Yes, the file naming convention from NOAA uses the hour of model runtime as the baseline hour 00, which doesn't necessarily mean 00:00 UTC. All hours are then indexed as an offset of 00, which is why we need to pull different "hours" depending on which model_cycle it's on. So yes, hours >= 24 will be subsequent days from the model run datetime. (But also for example, hour 10 of model_cycle hour 18 would be (18+10=28, and 28-24=4, so 4:00 UTC of the next day)

Reply:
I had hoped that the comment I added would clarify this, but if you have any suggestions for how I could revise my comment to make it clearer, please let me know!

Reply:
Ok thanks, clear now. You may have already outlined this in the backgrounder doc on our model runs, would it make sense to link to that doc in the comment here?

Reply:
oh boy. I totally forgot about that doc! I'll update that too to add NAM